### PR TITLE
[SPARK-40898][SQL] Quote function names in datatype mismatch errors

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to quote function names by `toSQLId()` that are passed to DataTypeMismatch as message parameters.

### Why are the changes needed?
To be consistent w/ other error messages.

### Does this PR introduce _any_ user-facing change?
Yes, it changes user-facing error message.

### How was this patch tested?
By running the modified test suites:
```
$ PYSPARK_PYTHON=python3 build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@LuciferYang @panbingkun Could you review this PR, please.

**Comment by LuciferYang** [comment]
The code change is fine to me, but it seems that some related tests need to be fixed

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/test/resources/sql-tests/results/ansi/interval.sql.out [hunk 1]
```diff
@@ -3609,7 +3609,7 @@ org.apache.spark.sql.AnalysisException
   "errorClass" : "DATATYPE_MISMATCH.DATA_DIFF_TYPES",
   "messageParameters" : {
     "dataType" : "(\"INTERVAL MONTH\" or \"INTERVAL DAY\")",
-    "functionName" : "function array",
+    "functionName" : "`function array`",
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/results/ansi/interval.sql.out:3612]: this seems wrong...

Comment (LuciferYang) [reply sql/core/src/test/resources/sql-tests/results/ansi/interval.sql.out:3612]: Should it be `function `array``?

**Comment by MaxGekk** [comment]
Merging to master. Thank you, @cloud-fan and @LuciferYang for review.

**Pull request closed without merging**